### PR TITLE
DB-5058: CLI docs and updated API reference

### PR DESCRIPTION
**Pull request by CobyPear**
## What changes were made?
- Update CONTRIBUTING.md with some more accurate info since the refactor
- Add docs regarding next-drupal-umami-addon
- Add docs explaining how to use the CLI in different contexts
- Regenerated API reference
## Where were the changes made?
- CONTRIBUTING.md
- Front-end docs
- API reference

## Additional information

Don't forget to [add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset) if needed!

### CODE REVIEW

**Review by backlineint — CHANGES_REQUESTED** [review]
Pushed up a coupled of small typo, formatting and link changes. Just take a look to make sure those make sense.

File: web/docs/Frontend Starters/Next.js/Next.js + Drupal/add-ons.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+title: 'Add-ons'
+slug: '/frontend-starters/nextjs/nextjs-drupal/add-ons'
+sidebar_position: 8
+---
+
+## Drupal Umami Demo Data Add-on
+
+This add-on adds necessary components and routes to your Pantheon Decoupled Kit
+Next.js Starter Kit to display data from Drupal's Umami demo site. This includes
+a route for `recipes`, as well as a link in the footer menu to the `recipes`
```
Comment (backlineint) [inline web/docs/Frontend Starters/Next.js/Next.js + Drupal/add-ons.md:12]: @CobyPear also noticed this after the fact. Double check me on this, but I don't think this is completely accurate. The add-on technically doesn't add a footer menu link - that just displays based on the data from Drupal. You could instead say something like 'This includes routes for all individual recipes, along with a index page that displays a grid of all recipes'

File: web/docs/Frontend Starters/create-pantheon-decoupled-kit.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+The CLI will not stop you from running any generators in any combination, but be
+aware that not all combination will result in a working application. If you run
+the CLI with multiple _project_ generators, you may end up with some unexpected
+results. These `project` generators should be used only in conjunction with
+`add-on` generators to avoid broken output.
+
+:::
+
+### Add-ons
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply web/docs/Frontend Starters/create-pantheon-decoupled-kit.md:62]: Open to this. When you say 'in a section above', do you mean a new section or the one right before this Add-ons section?

File: web/docs/Frontend Starters/Next.js/Next.js + Drupal/add-ons.md (hunk 1 above)
Comment (CobyPear) [reply web/docs/Frontend Starters/Next.js/Next.js + Drupal/add-ons.md:12]: Yes that sounds more accurate, good catch.

**Pull request merged**